### ISSUE TITLE
Import-Module issue: Assembly with the same name is already loaded

### ISSUE THREAD

**Issue by pmatthews05** [AzureAD/MSAL.PS#16]
Attempting to install your powershell module and unable to get it working.

I install the module using
```ps
Install-module -name MSAL.PS
```
I accepted the untrusted repository and accept license.

Then when I attempt to run a command 
```ps
Get-MsalToken
```
I get the error message
```text
Get-msaltoken: The 'Get-msaltoken' command was found in the module 'MSAL.PS', but the module could not be loaded. For more information, run 'Import-Module MSAL.PS'.
```
So I then attempt to import the module
```ps
Import-Module MSAL.PS -verbose
```
and the output is
```ps
VERBOSE: Loading module from path 'c:\Users\Me\Documents\powershell\Modules\MSAL.PS\4.16.0.3\MSAL.PS.psd1'.
VERBOSE: Loading 'Assembly' from path 'c:\Users\Me\Documents\powershell\Modules\MSAL.PS\4.16.0.3\Microsoft.Identity.Client.4.16.0\net45\Microsoft.Identity.Client.dll'.
VERBOSE: Loading 'Executable' from path 'c:\Users\Me\Documents\powershell\Modules\MSAL.PS\4.16.0.3\net45\Microsoft.Identity.Client.dll'.
VERBOSE: Loading 'Assembly' from path 'c:\Users\Me\Documents\powershell\Modules\MSAL.PS\4.16.0.3\Microsoft.Identity.Client.4.16.0\netcoreapp2.1\Microsoft.Identity.Client.dll'.
VERBOSE: Loading 'Executable' from path 'c:\Users\Me\Documents\powershell\Modules\MSAL.PS\4.16.0.3\Microsoft.Identity.Client.4.16.0\netcoreapp2.1\Microsoft.Identity.Client.dll'.
Import-Module: Assembly with same name is already loaded
```

I have attempted to install this on 3 different machines, attempted using PowerShell ISE, Powershell 7, and Powershell inside codespaces all return the same message. 

Note: PowerShell 7 was newly installed

Is there an issue, or am I doing something wrong?

Happy to provide more information if required.

**Comment by darrenjrobinson** [comment]
I'm seeing the same thing @pmatthews05 
@jasoth there seems to be an issue with the latest release 4.16.0.3 on PowerShell 7.0.x.

Module imports fine in Windows PowerShell

![image](https://user-images.githubusercontent.com/384733/90567778-09bbf380-e1ee-11ea-9e7f-81b96b51a50e.png)

Version 4.16.0.2 imports fine in PowerShell 7.0.3

![image](https://user-images.githubusercontent.com/384733/90567850-2526fe80-e1ee-11ea-96b0-9d6fdc90ca67.png)

BUT 4.16.0.3 bombs on import on PowerShell 7.0.3

![image](https://user-images.githubusercontent.com/384733/90567980-5d2e4180-e1ee-11ea-9013-14f300025bb0.png)

**Comment by jasoth** [comment]
Thank you both for reporting this. I have unlisted 4.16.0.3 from PS Gallery and rereleased with the fix in 4.16.0.4.

**Comment by darrenjrobinson** [comment]
Champion, thx @jasoth 
Working for me now @pmatthews05 with v 4.16.0.4 

![image](https://user-images.githubusercontent.com/384733/90586851-7c8f9380-e21b-11ea-941e-644fb4d42b50.png)

**Comment by pmatthews05** [comment]
Thank you @jasoth & @darrenjrobinson 

It is working for me now too with the new version v4.16.0.4.

**Comment by cobdev** [comment]
Hi everyone, 

It appears that this issue is happening again with MSAL.PS v4.37.0.0 in combination with ExchangeOnlineManagement v3.1.0 in PowerShell v7.3.4. I can reproduce this on multiple systems with this configuration.

Thanks for the help!

<img width="660" alt="2023-05-09_12-03" src="https://github.com/AzureAD/MSAL.PS/assets/10552026/7d603626-5eab-48fa-a836-20fb81a184b2">

**Comment by bchap1n** [comment]
> Thank you both for reporting this. I have unlisted 4.16.0.3 from PS Gallery and rereleased with the fix in 4.16.0.4.

excellent! I ran into this issue and was running the msal cmdlets in a Powershell Job as a workaround.
